Quarkus 3.3.1 jni error with CRT client:

Quarkus 3.3.1 behavior with CRT client:


```log
2023-09-04 10:02:30,049 ERROR [io.qua.sch.com.run.StatusEmitterInvoker] (executor-thread-1) Error occurred while executing task for trigger io.quarkus.quartz.runtime.QuartzSchedulerImpl$QuartzTrigger@7cdf89e0: java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: software.amazon.awssdk.crt.CrtRuntimeException.<init>(Ljava/lang/String;)V
       at java.base@17.0.8/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
       at java.base@17.0.8/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
       at java.base@17.0.8/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874)
       at java.base@17.0.8/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
       at java.base@17.0.8/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
       at java.base@17.0.8/java.util.concurrent.CompletableFuture$MinimalStage.whenComplete(CompletableFuture.java:2902)
       at io.quarkus.scheduler.common.runtime.DefaultInvoker.invoke(DefaultInvoker.java:24)
       at io.quarkus.scheduler.common.runtime.StatusEmitterInvoker.invoke(StatusEmitterInvoker.java:35)
       at io.quarkus.scheduler.common.runtime.SkipConcurrentExecutionInvoker.invoke(SkipConcurrentExecutionInvoker.java:37)
       at io.quarkus.quartz.runtime.QuartzSchedulerImpl$InvokerJob$1.handle(QuartzSchedulerImpl.java:875)
       at io.quarkus.quartz.runtime.QuartzSchedulerImpl$InvokerJob$1.handle(QuartzSchedulerImpl.java:871)
       at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
       at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
       at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
       at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
       at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
       at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
       at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
       at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
       at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
       at java.base@17.0.8/java.lang.Thread.run(Thread.java:833)
       at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:807)
       at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:210)
Caused by: java.lang.NoSuchMethodError: software.amazon.awssdk.crt.CrtRuntimeException.<init>(Ljava/lang/String;)V
       at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1267)
       at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.ThrowNew(JNIFunctions.java:878)
       at software.amazon.awssdk.crt.io.TlsContext.tlsContextNew(Native Method)
       at software.amazon.awssdk.crt.io.TlsContext.<init>(TlsContext.java:24)
       at software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient.<init>(AwsCrtAsyncHttpClient.java:103)
       at software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient.<init>(AwsCrtAsyncHttpClient.java:72)
       at software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient$DefaultBuilder.buildWithDefaults(AwsCrtAsyncHttpClient.java:400)
       at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.lambda$null$8(SdkDefaultClientBuilder.java:363)
       at java.base@17.0.8/java.util.Optional.map(Optional.java:260)
       at software.amazon.awssdk.utils.Either.lambda$map$0(Either.java:51)
       at java.base@17.0.8/java.util.Optional.orElseGet(Optional.java:364)
       at software.amazon.awssdk.utils.Either.map(Either.java:51)
       at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.lambda$resolveAsyncHttpClient$9(SdkDefaultClientBuilder.java:363)
       at java.base@17.0.8/java.util.Optional.map(Optional.java:260)
       at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.resolveAsyncHttpClient(SdkDefaultClientBuilder.java:363)
       at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.finalizeAsyncConfiguration(SdkDefaultClientBuilder.java:303)
       at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.asyncClientConfiguration(SdkDefaultClientBuilder.java:209)
       at software.amazon.awssdk.services.s3.DefaultS3AsyncClientBuilder.buildClient(DefaultS3AsyncClientBuilder.java:54)
       at software.amazon.awssdk.services.s3.DefaultS3AsyncClientBuilder.buildClient(DefaultS3AsyncClientBuilder.java:30)
       at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.build(SdkDefaultClientBuilder.java:147)
       at software.amazon.awssdk.services.s3.S3AsyncClientBuilder_48f892d2989fb9718e1064848c00fe33d3c8cd8a_Synthetic_ClientProxy.build(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer.<init>(S3ClientProducer.java:25)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_Bean.doCreate(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_Bean.create(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_Bean.create(Unknown Source)
       at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
       at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
       at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
       at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
       at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
       at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
       at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_ClientProxy.arc$delegate(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_ClientProxy.arc_contextualInstance(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_asyncClient_456f249176f7d1a9150792617e095fe59674e529_Bean.doCreate(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_asyncClient_456f249176f7d1a9150792617e095fe59674e529_Bean.create(Unknown Source)
       at io.quarkus.amazon.s3.runtime.S3ClientProducer_ProducerMethod_asyncClient_456f249176f7d1a9150792617e095fe59674e529_Bean.create(Unknown Source)
       at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
       at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
       at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
       at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
       at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
       at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
       at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
       at software.amazon.awssdk.services.s3.S3ClientProducer_ProducerMethod_asyncClient_456f249176f7d1a9150792617e095fe59674e529_ClientProxy.arc$delegate(Unknown Source)
       at software.amazon.awssdk.services.s3.S3ClientProducer_ProducerMethod_asyncClient_456f249176f7d1a9150792617e095fe59674e529_ClientProxy.listObjects(Unknown Source)
       at app.yodha.backend.bloomreachintegration.imp.ImportService.import(ImportService.kt:22)
       at app.yodha.backend.bloomreachintegration.imp.ImportService_ClientProxy.import(Unknown Source)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventService.import(ImportEventService.kt:36)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventService_ClientProxy.import(Unknown Source)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventJob$import$2.invoke(ImportEventJob.kt:21)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventJob$import$2.invoke(ImportEventJob.kt:20)
       at app.yodha.backend.bloomreachintegration.util.UtilsKt.runLoggingDuration(Utils.kt:9)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventJob.import(ImportEventJob.kt:20)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventJob_ClientProxy.import(Unknown Source)
       at app.yodha.backend.bloomreachintegration.imp.event.ImportEventJob_ScheduledInvoker_import_1d75079875fcc70c9a44eb2b7e2626d0febae4a8.invokeBean(Unknown Source)
       ... 17 more


```

_Originally posted by @vladimirfx in https://github.com/quarkiverse/quarkus-amazon-services/issues/860#issuecomment-1704985842_

--- a/crt/deployment/src/main/java/io/quarkiverse/amazonservices/aws/crt/deployment/AwsCrtProcessor.java
+++ b/crt/deployment/src/main/java/io/quarkiverse/amazonservices/aws/crt/deployment/AwsCrtProcessor.java
@@ -110,6 +110,9 @@ public class AwsCrtProcessor {
                 "software.amazon.awssdk.crt.CrtResource", "releaseReferences"));
         jniRuntimeAccessMethod.produce(new JniRuntimeAccessMethodBuildItem(
                 "software.amazon.awssdk.crt.CrtRuntimeException", "<init>", "int"));
+        // for exception thrown from https://github.com/awslabs/aws-crt-java/blob/3eea28d5f617f408655054efd33278ce84d0563b/src/native/crt.c#L258-L259
+        jniRuntimeAccessMethod.produce(new JniRuntimeAccessMethodBuildItem(
+                "software.amazon.awssdk.crt.CrtRuntimeException", "<init>", "java.lang.String"));
         jniRuntimeAccessField.produce(new JniRuntimeAccessFieldBuildItem(
                 "software.amazon.awssdk.crt.CrtRuntimeException", "errorCode"));
         jniRuntimeAccessMethod.produce(new JniRuntimeAccessMethodBuildItem(

